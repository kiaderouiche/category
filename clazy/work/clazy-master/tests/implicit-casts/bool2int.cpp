#include <QtCore/QDebug>

struct S {
    void operator=(int);
};

void foo(int) {}

extern "C" void takesInt_C(int);
extern void takesInt_Cpp(int);

struct A
{
    void foo(int);
};

void testIntToBool(bool b1)
{
    bool b2;
    foo(false); // Warn
    foo(b1); // Warn
    foo(b2); // Warn
    A a;
    a.foo(false); // Warn

    takesInt_C(false); // OK
    takesInt_Cpp(false); // Warn

    S s; s = false; // OK
    qDebug("Foo %d", false); // OK
}
