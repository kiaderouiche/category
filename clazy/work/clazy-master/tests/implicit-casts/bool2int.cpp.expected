implicit-casts/bool2int.cpp:20:9: warning: Implicit bool to int cast (argument 1) [-Wclazy-implicit-casts]
implicit-casts/bool2int.cpp:21:9: warning: Implicit bool to int cast (argument 1) [-Wclazy-implicit-casts]
implicit-casts/bool2int.cpp:22:9: warning: Implicit bool to int cast (argument 1) [-Wclazy-implicit-casts]
implicit-casts/bool2int.cpp:24:11: warning: Implicit bool to int cast (argument 1) [-Wclazy-implicit-casts]
implicit-casts/bool2int.cpp:27:18: warning: Implicit bool to int cast (argument 1) [-Wclazy-implicit-casts]
